tests/rptest/services/redpanda.py

```diff
@@ -949,27 +949,23 @@ def start_si(self):
 
         self.logger.debug(
             f"Creating S3 bucket: {self._si_settings.cloud_storage_bucket}")
-        self._s3client.create_bucket(self._si_settings.cloud_storage_bucket)
+        self.s3_client.create_bucket(self._si_settings.cloud_storage_bucket)
 
     def list_buckets(self) -> dict[str, Union[list, dict]]:
-        assert self._s3client is not None
-        return self._s3client.list_buckets()
-
-    @property
-    def s3_client(self):
-        return self._s3client
+        assert self.s3_client is not None
+        return self.s3_client.list_buckets()
 
     def delete_bucket_from_si(self):
-        assert self._s3client is not None
+        assert self.s3_client is not None
```

Comment:
Should we have additional messages around this (and other newly added) asserts? It makes it easier to understand the error in buildkite output.

Reply:
These shouldn't be firing; they are cases that don't happen in practice (or else we'd have other stack traces).  I prefer to keep these concise. It is clear, to me, what the intention is.